File: typescript/practice/classic_puzzle/easy/the-descent.ts
/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/


// game loop
let targetId: number;
do {
    const mountainHs: number[] = [];
    for (let i = 0; i < 8; i++) {
        const mountainH: number = parseInt(readline()); // represents the height of one mountain.

        mountainHs.push(mountainH);
    }

    // Write an action using console.log()
    // To debug: console.error('Debug messages...');

    let targetH: number = 0;
    targetId = null;
    for (let i: number = 0; i < mountainHs.length; i++) {
        if (targetH < mountainHs[i]) {
            targetH = mountainHs[i];
            targetId = i;
        }
    }

    // console.log('4');     // The index of the mountain to fire on.
    console.log(targetId);
} while (targetId !== null);

File: ruby/practice/classic_puzzle/easy/the-descent.rb
$stdout.sync = true # DO NOT REMOVE
# The while loop represents the game.
# Each iteration represents a turn of the game
# where you are given inputs (the heights of the mountains)
# and where you have to print an output (the index of the mountain to fire on)
# The inputs you are given are automatically updated according to your last actions.

# game loop
loop do
  target_h  = 0
  target_id = 0

  8.times do |i|
    mountain_h = gets.to_i # represents the height of one mountain
    if target_h < mountain_h
      target_h = mountain_h
      target_id = i
    end
  end

  # Write an action using puts
  # To debug: STDERR.puts "Debug messages

  # puts target_id # The index of the mountain to fire on.
  puts target_id
end
